internal/transport/http2_client.go

```diff
@@ -231,9 +231,22 @@ func newHTTP2Client(connectCtx, ctx context.Context, addr resolver.Address, opts
 		contextWithHandshakeInfo := internal.NewClientHandshakeInfoContext.(func(context.Context, credentials.ClientHandshakeInfo) context.Context)
 		connectCtx = contextWithHandshakeInfo(connectCtx, credentials.ClientHandshakeInfo{Attributes: addr.Attributes})
 		conn, authInfo, err = transportCreds.ClientHandshake(connectCtx, addr.ServerName, conn)
+		type internalInfo interface {
```

Comment:
Hmmmmm, maybe `CheckSecurityLevel` should accept an `AuthInfo` instead of a `Context`.  Does this cause too many changes?

Reply:
Sure, consider it done!